### ISSUE TITLE
[a11y]: Search [Expanded] has a11y Violation - Form control with "button" role has no associated label

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#18202]
### Package

@carbon/react

### Package version

1.72.0

### Automated testing tool and ruleset

IBM Accessibility Checker latest deployment and 13Nov24 deployment

### Description

Expandable search component has a11y Violation 
![Image](https://github.com/user-attachments/assets/ddd507e7-a877-4814-bccc-eecbec926fa7)

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=&globals=&id=components-search--expandable&viewMode=story

### Steps to reproduce

Open storybook 
visit expandable search storybook - https://react.carbondesignsystem.com/iframe.html?args=&globals=&id=components-search--expandable&viewMode=story
run Accessibility Checker with latest deployment 

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
